### PR TITLE
feat: prorated_service_period and estimated_full_service_period

### PR DESCRIPTION
**Pull request by esoubihe**
Pull Request
**Are you a current Fivetran customer?** 

**What change(s) does this PR introduce?** 

**Did you update the CHANGELOG?** 

- [ ] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [ ] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [ ] No 

**How did you test the PR changes?** 

 

- [ ] CircleCi  
- [ ] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [ ] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/stripe__invoice_line_items.sql [hunk 1]
```diff
@@ -64,7 +64,10 @@ select
     customer.customer_id
 
     {% if var('using_subscriptions', True) %}
-    ,subscription.subscription_id,
+    ,invoice_line_item.proration, 
+    extract(epoch from (invoice_line_item.period_end::timestamp - invoice_line_item.period_start::timestamp + (plan.interval_count || ' MONTH')::INTERVAL)) as estimated_service_period_start,
```
Comment (fmluizao) [inline models/stripe__invoice_line_items.sql:68]: shouldn't we consider `plan_interval` here?

```suggestion
    extract(epoch from (invoice_line_item.period_end::timestamp - invoice_line_item.period_start::timestamp + (plan.interval_count || ' ' || plan.plan_interval)::INTERVAL)) as estimated_service_period_start,
```

Comment (esoubihe) [reply models/stripe__invoice_line_items.sql:68]: `plan_interval` is a string unless we transform it to integer

Comment (fmluizao) [reply models/stripe__invoice_line_items.sql:68]: My suggestion is change

```
 (plan.interval_count || ' MONTH')::INTERVAL
```

to

```
 (plan.interval_count || ' ' || plan.plan_interval)::INTERVAL
```

Which will be evaulated as '1 month', or '1 year', etc.

Comment (fmluizao) [reply models/stripe__invoice_line_items.sql:68]: Also, I believe `estimated_full_service_period` is wrong. I think it should be something like this:

```sql
case 
  when invoice_line_item.proration then 
     extract(epoch from (invoice_line_item.period_end - invoice_line_item.period_start)) as estimated_full_service_period
  else
     --full duration of subscription
end
```

Comment (esoubihe) [reply models/stripe__invoice_line_items.sql:68]: Actually, it's the inverse. When an item is a proration, we need to calculate the full period by looking at `period_end - plan_interval`, and when it's not a proration, do what you said above. But you need to revert the order and put period_start before period_end

**Comment by esoubihe** [comment]
Superseded by #4

**Pull request closed without merging**